subcommand/consul-init/command.go

```diff
@@ -0,0 +1,162 @@
+package consulinit
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"os/signal"
+	"strings"
+	"sync"
+	"sync/atomic"
+	"syscall"
+	"time"
+
+	"github.com/hashicorp/consul-k8s/consul"
+	"github.com/hashicorp/consul-k8s/subcommand/common"
+	"github.com/hashicorp/consul-k8s/subcommand/flags"
+	"github.com/hashicorp/consul/api"
+	"github.com/mitchellh/cli"
+)
+
+const bearerTokenFile = "/var/run/secrets/kubernetes.io/serviceaccount/token"
+const tokenSinkFile = "/consul/connect-inject/acl-token"
+
+type Command struct {
+	UI cli.Ui
+
+	flagACLAuthMethod   string // Auth Method to use for ACLs, if enabled
+	flagMeta            string // Flag for metadata to consul login.
+	flagBearerTokenFile string // Location of the bearer token.
+	flagTokenSinkFile   string // Location to write the output token.
+	numACLLoginRetries  int    // Number of times to attempt to retry ACL().Login(), default: 3.
+
+	flagSet *flag.FlagSet
+	http    *flags.HTTPFlags
+
+	consulClient *api.Client
+
+	sigCh chan os.Signal
+	once  sync.Once
+	help  string
+	cert  atomic.Value
+}
+
+func (c *Command) init() {
+	c.flagSet = flag.NewFlagSet("", flag.ContinueOnError)
+	c.flagSet.StringVar(&c.flagACLAuthMethod, "method", "",
+		"The name of the Kubernetes Auth Method to use for connectInjection if ACLs are enabled.")
+	c.flagSet.StringVar(&c.flagMeta, "meta", "",
+		"Metadata to be passed to consul login api.")
+	c.flagSet.StringVar(&c.flagBearerTokenFile, "bearer-token-file", bearerTokenFile,
+		"Token file to be used for consul login api.")
+	c.flagSet.StringVar(&c.flagTokenSinkFile, "token-sink-file", tokenSinkFile,
+		"Token sink file to be used for consul login api.")
+
+	c.http = &flags.HTTPFlags{}
+
+	flags.Merge(c.flagSet, c.http.Flags())
+	c.help = flags.Usage(help, c.flagSet)
+
+	// Default number of times to attempt ACL().Login()
+	c.numACLLoginRetries = 3
+
+	// Wait on an interrupt or terminate for exit, be sure to init it before running
+	// the controller so that we don't receive an interrupt before it's ready.
+	if c.sigCh == nil {
+		c.sigCh = make(chan os.Signal, 1)
+		signal.Notify(c.sigCh, syscall.SIGINT, syscall.SIGTERM)
+	}
+}
+
+func (c *Command) Run(args []string) int {
+	c.once.Do(c.init)
+	var err error
+	if err := c.flagSet.Parse(args); err != nil {
+		return 1
+	}
+
+	// Validate flags.
+	if c.flagACLAuthMethod == "" {
+		c.UI.Error("-method must be set")
+		return 1
+	}
+	if c.flagMeta == "" {
```

Comment:
So this command is not generally consumable by users, it won't do anything meaningful standalone, meta will always be set outside of tests in the current incarnation of the code. However..  I could wrap these 4 validation checks inside of some sort if `if c.flagACLMethod {       }` ?
I don't think its a massive overhaul of the code or tests to leave these as is for current PR since it ONLY functions as a means to fetch ACLs.
Thoughts ?